stock_available/models/product_product.py

```diff
@@ -16,54 +16,35 @@ class ProductProduct(models.Model):
     _inherit = 'product.product'
 
     @api.multi
-    def _product_available(self, field_names=None, arg=False):
-        res = super(ProductProduct, self)._product_available(
-            field_names=field_names, arg=arg)
-        for prod_id in res:
-            res[prod_id]['immediately_usable_qty'] = res[prod_id][
-                'virtual_available']
-            res[prod_id]['potential_qty'] = 0.0
-
+    def _compute_available_quantities_dict(self):
+        res = {}
+        for product in self:
+            res[product.id] = {}
+            res[product.id]['immediately_usable_qty'] = \
+                product.virtual_available
+            res[product.id]['potential_qty'] = 0.0
         return res
 
     @api.multi
     @api.depends('virtual_available')
-    def _compute_immediately_usable_qty(self):
-        """No-op implementation of the stock available to promise.
-
-        By default, available to promise = forecasted quantity.
-
-        **Each** sub-module **must** override this method in **both**
-            `product.product` **and** `product.template`, because we can't
-            decide in advance how to compute the template's quantity from the
-            variants.
-        """
-        res = self._product_available()
-        for prod in self:
-            prod.immediately_usable_qty = res[prod.id][
-                'immediately_usable_qty']
-
-    @api.multi
-    @api.depends()
-    def _compute_potential_qty(self):
-        """Set potential qty to 0.0 to define the field defintion used by
-        other modules to inherit it
-        """
-        res = self._product_available()
-        for prod in self:
-            prod.immediately_usable_qty = res[prod.id][
-                'potential_qty']
+    def _compute_available_quantities(self):
```

Comment:
It looks like this is duplicated in `product.template` why not just escalate to the template method and just pass in the `product.product` instead? Something like:

```
self.env['product.template']. _compute_available_quantities(self)
```

Admittedly I haven't tried it in this context, but I'm almost positive I've done something similar elsewhere without triggering an apples to oranges error.